tensorflow_addons/layers/normalizations.py

```diff
@@ -316,3 +316,225 @@ def __init__(self, **kwargs):
+        self.beta_regularizer = tf.keras.regularizers.get(beta_regularizer)
+        self.gamma_regularizer = tf.keras.regularizers.get(gamma_regularizer)
+        self.tau_regularizer = tf.keras.regularizers.get(tau_regularizer)
+        self.beta_constraint = tf.keras.constraints.get(beta_constraint)
+        self.gamma_constraint = tf.keras.constraints.get(gamma_constraint)
+        self.tau_constraint = tf.keras.constraints.get(tau_constraint)
+        self.eps_var_initializer = tf.keras.initializers.get(
+            eps_var_initializer)
+        self.supports_masking = True
+        if isinstance(axis, list):
+            self.axis = axis[:]
+        else:
+            raise TypeError('axis must be list, type given: %s' % type(axis))
+        self._check_axis()
+
+    def build(self, input_shape):
+        self._check_if_input_shape_is_none(input_shape)
+        self._create_input_spec(input_shape)
+        self._add_gamma_weight(input_shape)
+        self._add_beta_weight(input_shape)
+        self._add_tau_weight(input_shape)
+        self._add_eps_var_weight()
+        self.built = True
+        super(FilterResponseNormalization, self).build(input_shape)
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+    def get_config(self):
+        config = {
+            'axis':
+            self.axis,
+            'epsilon':
+            self.epsilon,
+            'beta_initializer':
+            tf.keras.initializers.serialize(self.beta_initializer),
+            'gamma_initializer':
+            tf.keras.initializers.serialize(self.gamma_initializer),
+            'tau_initializer':
+            tf.keras.initializers.serialize(self.tau_initializer),
+            'eps_var_initializer':
+            tf.keras.initializers.serialize(self.eps_var_initializer),
+            'beta_regularizer':
+            tf.keras.regularizers.serialize(self.beta_regularizer),
+            'gamma_regularizer':
+            tf.keras.regularizers.serialize(self.gamma_regularizer),
+            'tau_regularizer':
+            tf.keras.regularizers.serialize(self.tau_regularizer),
+            'beta_constraint':
+            tf.keras.constraints.serialize(self.beta_constraint),
+            'gamma_constraint':
+            tf.keras.constraints.serialize(self.gamma_constraint),
+            'tau_constraint':
+            tf.keras.constraints.serialize(self.tau_constraint)
+        }
+        base_config = super(FilterResponseNormalization, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+    def call(self, inputs):
+        # Compute the mean norm of activations per channel.
+        nu2 = tf.reduce_mean(tf.square(inputs), axis=self.axis, keepdims=True)
+        # Perform FRN.
+        inputs = inputs * tf.math.rsqrt(nu2 + tf.math.abs(self.epsilon))
+        # Return after applying the Offset-ReLU non-linearity.
+        return tf.maximum(self.gamma * inputs + self.beta,
```

Comment:
It would be better to make the activation configurable so that users can experiment with other combinations. While the paper does demonstrate that FRN+TLU combination works well in a variety of settings, in future we may find that a different combination works even better. Making the activation configurable will future proof the layer.

As a result I would suggest making TLU a separate activation layer similar to ReLU, PReLU etc. This way it can be swapped in/out. For example, I found it useful to replace **all** activations in a network with TLU instead of just those that followed normalization (e.g. in ResnetV1 backbone model used in object detection in paper)..

Reply:
That actually makes sense. Okay, I will implement FRN and TLU as separate layers then